lib/schema/hooks.go

```diff
@@ -77,6 +77,12 @@ type Hooks struct {
 	Backup *Hook `json:"backup,omitempty"`
 	// Restore restores application state from a backup
 	Restore *Hook `json:"restore,omitempty"`
+	// OverlayInstall is a hook for installing a custom overlay network
+	OverlayInstall *Hook `json:"overlayInstall,omitempty"`
```

Comment:
I wonder if `networkInstall` would be a better choice of word (since "overlay" can refer not just to network but e.g. to storage driver too so may lead to confusion). Same for other hooks.